File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.py
class Solution(object):
    def maxAbsoluteSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mx=mn=maxsum=minsum=0
        for x in nums:
            maxsum+=x
            minsum+=x
            mx=max(mx,maxsum)
            mn=min(mn,minsum)
            if minsum>0: minsum=0
            if maxsum<0: maxsum=0
        return max(mx,abs(mn)) 
        

File: 1849-maximum-absolute-sum-of-any-subarray/maximum-absolute-sum-of-any-subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int min=0,max=0,maxsum=0,minsum=0;
        for(int num:nums){
            maxsum+=num;
            minsum+=num;
            max=Math.max(max,maxsum);
            min=Math.min(min,minsum);
            if(maxsum<0) maxsum=0;
            if(minsum>0) minsum=0;
        }
        return Math.max(max,Math.abs(min));
    }
}